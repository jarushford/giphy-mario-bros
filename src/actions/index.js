export const setPlayers = (players) => ({
  type: 'SET_PLAYERS',
  players
})

export const clearPlayers = () => ({
  type: 'CLEAR_PLAYERS'
})

export const setError = (error) => ({
  type: 'SET_ERROR',
  error
})

export const clearError = () => ({
  type: 'CLEAR_ERROR'
})