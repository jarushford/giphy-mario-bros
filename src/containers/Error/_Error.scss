.error {
  @include main();

  h1 {
    font-size: 2rem;
    color: $white;
    text-shadow: $outlinethick;
    text-align: center;
    font-family: 'Mario';
  }

  button {
    @include btn();
    width: 300px;
    cursor: pointer;
  }
}