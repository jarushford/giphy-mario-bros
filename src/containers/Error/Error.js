import React from 'react'
import { clearError } from '../../actions'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

export function Error({ clearError }) {
  return (
    <section className="error">
      <h1 className="error-msg">Something has gone terribly wrong.</h1>
      <Link to='/' onClick={clearError}>
        <button>Return to Site</button>
      </Link>
    </section>
  )
}

export const mapDispatchToProps = (dispatch) => ({
  clearError: () => dispatch(clearError())
})

export default connect(null, mapDispatchToProps)(Error)

Error.propTypes = {
  clearError: PropTypes.func.isRequired
}
