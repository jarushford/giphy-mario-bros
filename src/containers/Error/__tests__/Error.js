import React from 'react'
import { Error, mapDispatchToProps } from '../Error'
import { shallow } from 'enzyme'
import { clearError } from '../../../actions'

describe('Error', () => {
  let mockClear

  beforeEach(() => {
    mockClear = jest.fn()
  })

  it('should match the snapshot', () => {
    const wrapper = shallow(<Error clearError={mockClear} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should clear error on click', () => {
    const wrapper = shallow(<Error clearError={mockClear} />)

    wrapper.find('Link').simulate('click')

    expect(mockClear).toBeCalled()
  })

  it('should dispatch clearError when the prop clearError is called', () => {
    const mockDispatch = jest.fn()
    const expected = clearError()

    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.clearError()

    expect(mockDispatch).toBeCalledWith(expected)
  })
})