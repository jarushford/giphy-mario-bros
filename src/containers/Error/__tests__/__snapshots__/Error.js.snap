// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Error
    clearError={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="error-msg"
        >
          Something has gone terribly wrong.
        </h1>,
        <Link
          onClick={[MockFunction]}
          replace={false}
          to="/"
        >
          <button>
            Return to Site
          </button>
        </Link>,
      ],
      "className": "error",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Something has gone terribly wrong.",
          "className": "error-msg",
        },
        "ref": null,
        "rendered": "Something has gone terribly wrong.",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button>
            Return to Site
          </button>,
          "onClick": [MockFunction],
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Return to Site",
          },
          "ref": null,
          "rendered": "Return to Site",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="error-msg"
          >
            Something has gone terribly wrong.
          </h1>,
          <Link
            onClick={[MockFunction]}
            replace={false}
            to="/"
          >
            <button>
              Return to Site
            </button>
          </Link>,
        ],
        "className": "error",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Something has gone terribly wrong.",
            "className": "error-msg",
          },
          "ref": null,
          "rendered": "Something has gone terribly wrong.",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button>
              Return to Site
            </button>,
            "onClick": [MockFunction],
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Return to Site",
            },
            "ref": null,
            "rendered": "Return to Site",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
